### PR TITLE
feat: typescript.tsconfigPath array support

### LINKED ISSUE #305: Add array support for typescript.tsconfigPath
### Clear and concise description of the problem

You may have more than one tsconfig in your project. 
For example vite default template has 'tsconfig.json' and 'tsconfig.node.json'.
To enable 'type aware rules' properly you need to specify both. 
ESLint parserOptions.project supports array for this - [LINK](https://typescript-eslint.io/linting/typed-linting/monorepos/#one-tsconfigjson-per-package-and-an-optional-one-in-the-root)

### Suggested solution

Will make a PR

### Alternative

_No response_

### Additional context

Related: 
https://github.com/antfu/eslint-config/issues/265
https://github.com/antfu/eslint-config/issues/286

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by rost-git**
### Description

Multiple tsconfigs support
```ts
typescript: {
  tsconfigPath: ['tsconfig.json', 'tsconfig.node.json']
}
```

### Linked Issues
close #305

### CODE REVIEW

File: src/configs/typescript.ts [hunk 1]
```diff
@@ -36,6 +36,12 @@ export function typescript(
     'ts/unbound-method': 'error',
   }
 
+  const tsconfigPath = typeof tsc === 'string' && tsc.trim()
```
Comment (antfu) [inline src/configs/typescript.ts:39]: Why `trim` is needed?

Comment (rost-git) [reply src/configs/typescript.ts:39]: empty string check

File: README.md [hunk 2]
```diff
@@ -328,6 +328,8 @@ import antfu from '@antfu/eslint-config'
 export default antfu({
   typescript: {
     tsconfigPath: 'tsconfig.json',
+    // or if you have multiple tsconfigs
+    tsconfigPath: ['tsconfig.json', 'tsconfig.node.json'],
```
Comment (Dimava) [inline README.md:332]: Copypasted code should "just work", so this line should be commented

It supports globs so I'd recommend adding `"packages/*/tsconfig.json"` here to imply that
And possibly `"tsconfig.eslint.json"` to imply you can make one especially for linting

File: src/configs/typescript.ts (hunk 1 above)
Comment (Dimava) [reply src/configs/typescript.ts:39]:
```suggestion
  const tsconfigPath = typeof tsc === 'string' ? (
     tsc.trim() !== '' ? [tsc] : null
   ) : ...
```

Comment (antfu) [reply src/configs/typescript.ts]: I pushed a change. I think that setting `tsconfigPath: " "` should be a user error, and we shouldn't tolerant that implicitly.

Comment (rost-git) [reply src/configs/typescript.ts]: agree, thanks
_just a habit to deal with user input - but here we have 'advanced' users_

**Pull request merged**